app/views/questionnaire.py

```diff
@@ -303,20 +356,52 @@ def submit_answers(routing_path, eq_id, form_type):
         if not sent:
             raise SubmissionFailedException()
 
-        get_questionnaire_store(current_user.user_id, current_user.user_ik).delete()
+        store_submitted_responses_and_session_data(answer_store.answers, metadata)
```

Comment:
I don't think we should be combining this into one method. We always store the session data, but only store the submitted responses if it's enabled for the questionnaire.